src/Compilers/CSharp/Test/Semantic/Semantics/UsingDeclarationTests.cs

```diff
@@ -0,0 +1,508 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.CSharp.UnitTests;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.Semantic.UnitTests.Semantics
+{
+    /// <summary>
+    /// Tests related to binding (but not lowering) using declarations (i.e. using var x = ...).
+    /// </summary>
+    public class UsingDeclarationTests : CompilingTestBase
+    {
+        [Fact]
+        public void UsingVariableIsNotReportedAsUnused()
+        {
+            var source = @"
+using System;
+class C
+{
+    static void Main()
+    {
+        using var x = (IDisposable)null;
+    }
+}
+";
+            CreateCompilation(source).VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void DisallowGoToForwardAcrossUsingDeclarations()
+        {
+            var source = @"
+using System;
+class C
+{
+    static void Main()
+    {
+        goto label1;
+        using var x = (IDisposable)null;
+
+        label1:
+        return;
+    }
+}
+";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (7,9): error CS8641: A goto target within the same block can not cross a using declaration.
```

Comment:
>A goto target within the same block can not cross a using declaration. [](start = 40, length = 70)

Consider updating comment text to match resource for `ERR_GoToForwardJumpOverUsingVar` and `ERR_GoToBackwardJumpOverUsingVar`.

Reply:
I also have an open issue (https://github.com/dotnet/roslyn/issues/28892) tracking the error messages. I'm going to do a full clean up pass before merging, as some of the async streams work has an impact on them